src/diffusers/pipeline_utils.py

```diff
@@ -373,6 +374,11 @@ def from_pretrained(cls, pretrained_model_name_or_path: Optional[Union[str, os.P
             if custom_pipeline is not None:
                 allow_patterns += [CUSTOM_PIPELINE_FILE_NAME]
 
+            requested_pipeline_class = config_dict.get("_class_name", cls.__name__)
```

Comment:
This should cover all cases, right?

Reply:
Yes that'll be a very powerful metric - nice!